flowetl/etl/tests/test_render_sql_callable.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import patch, MagicMock
+from pathlib import Path
+
+from airflow import DAG
+from airflow.operators import BaseOperator
+
+from etl.production_task_callables import render_sql_callable
+
+
+class MockDagRun:
+    def __init__(self, conf):
+        self.conf = conf
+
+
+def test_render_sql_callable(tmpdir):
+    """
+    Test that the render sql callable is able to construct the
+    correct sql from a template in the correct location and issues
+    this sql to the db_hook run command.
+    """
+
+    # Make temp sql template in "config/etl/voice/init.sql" where
+    # config is a temp dir
+
+    config_dir = tmpdir.mkdir("config")
+    etl_dir = config_dir.mkdir("etl")
+    voice_dir = etl_dir.mkdir("voice")
+
+    template_name = "init"
+    init_template_path = voice_dir.join(f"{template_name}.sql")
+    init_template = "select {{number}}"
+    init_template_path.write(init_template)
+
+    # Mocks so we can use the templating feature of the operator this
```

Comment:
This is a slightly nitpicky comment, but in the testing world there is actually a fairly specific definition for what a "mock" is (as opposed to other kinds of test doubles, such as dummy/fake/stub/spy/...). Martin Fowler explains some of the differences [here](https://martinfowler.com/articles/mocksArentStubs.html#TheDifferenceBetweenMocksAndStubs) (and the always awesome Sandi Metz also gave a great [talk](https://www.youtube.com/watch?v=URSWYvyc42M) which clarified the difference for me - mocks are a tool to test "outgoing command messages"; see her chart at 27:45 in the talk).

Technically the only mock in this test is the `mock_pghook` because this is the one you use to test the expectation that a particular method call that should have occurred. The other variables just contain just drop-in replacements for real objects to provide scaffolding that is needed to set up the test and make it work.

So to make the test slightly easier to read I'd personally rename the first two variables (as well as `MockDagRun` above) so that they don't contain the term "mock" (because if they do, as the reader of the test I would expect them to play a part in the assert statement below, which isn't the case and confused me a little at first).

I guess technically they are stub or fake objects, but I'd be happy to name them "dummy" as well. So perhaps `DummyDagRun`/`dummy_dag`/`dummy_task_op` or `FakeDagRun`/`fake_dag`/`fake_task_op` or similar?